pkgs/applications/graphics/ufraw/default.nix

```diff
@@ -1,38 +1,72 @@
-{ fetchurl, stdenv, pkgconfig, gtk2, gettext, bzip2, zlib
-, withGimpPlugin ? true, gimp ? null
-, libjpeg, libtiff, cfitsio, exiv2, lcms2, gtkimageview, lensfun }:
+{ stdenv
+, fetchFromGitHub
+
+, autoconf
+, automake
+, bzip2
+, cfitsio
+, exiv2
+, gettext
+, gimp ? null
+, gtk2
+, gtkimageview
+, lcms2
+, lensfun
+, libjpeg
+, libtiff
+, perl
+, pkgconfig
+, zlib
+
+, withGimpPlugin ? true
+}:
 
 assert withGimpPlugin -> gimp != null;
 
 stdenv.mkDerivation rec {
-  name = "ufraw-0.22";
+  pname = "ufraw";
+  version = "unstable-2019-06-12";
 
-  src = fetchurl {
-    # XXX: These guys appear to mutate uploaded tarballs!
-    url = "mirror://sourceforge/ufraw/${name}.tar.gz";
-    sha256 = "0pm216pg0vr44gwz9vcvq3fsf8r5iayljhf5nis2mnw7wn6d5azp";
+  src = fetchFromGitHub {
+    owner = "sergiomb2";
+    repo = "ufraw";
+    rev = "c65b4237dcb430fb274e4778afaf5df9a18e04e6";
+    sha256 = "02icn67bsinvgliy62qa6v7gmwgp2sh15jvm8iiz3c7g1h74f0b7";
   };
 
   outputs = [ "out" ] ++ stdenv.lib.optional withGimpPlugin "gimpPlugin";
 
-  nativeBuildInputs = [ pkgconfig gettext ];
+  nativeBuildInputs = [ autoconf automake gettext perl pkgconfig ];
+
   buildInputs = [
-    gtk2 gtkimageview bzip2 zlib
-    libjpeg libtiff cfitsio exiv2 lcms2 lensfun
+    bzip2
+    cfitsio
+    exiv2
+    gtk2
+    gtkimageview
+    lcms2
+    lensfun
+    libjpeg
+    libtiff
+    zlib
   ] ++ stdenv.lib.optional withGimpPlugin gimp;
 
+  preConfigure = "./autogen.sh";
+
   configureFlags = [
-    "--enable-extras"
-    "--enable-dst-correction"
     "--enable-contrast"
+    "--enable-dst-correction"
+    "--enable-extras"
   ] ++ stdenv.lib.optional withGimpPlugin "--with-gimp";
 
-  postInstall = stdenv.lib.optionalString withGimpPlugin ''
+  postInstall = ''
+    rm -f $out/bin/dcraw
```

Comment:
Why are you deleting this file? Could you add a simple comment here and a more detailed description into the commit message?

Reply:
The problem is the `--enable-extras` option. From the README:
```
--enable-extras: build the extra binaries - dcraw, nikon-curve.
```
```
For Packagers
=============
You are free to package UFRaw in any way, as long as it complies with the
GNU GPL. The followings are just recommendations.

Do not package the executables generated by by --enable-extras. These extras
are there for testing the code during development. They are of no interest
to end user. Specifically, if you want to package dcraw, you should use
Dave's original code and not UFRaw's modified code.
```
However, the existing derivation was already compiling with this option; I took this to mean that somebody wanted the `nikon-curve` binary for some reason, so I left it in to be conservative.

Checking the commit history, I see it was you that added the option :) so if you're not aware of any use case, I think it would be better to remove `--enable-extras` and see if anyone complains.

Reply:
I merely copied the flags from https://github.com/NixOS/nixpkgs/commit/63a234c4a720f3457801043276e1884e451d3b85#diff-9120f8657759d45707b61eaf16c44e4bL188-L196. The [original packager](https://github.com/NixOS/nixpkgs/commit/b241732d8cd5c6b22bfd8f2a256d33790929f5c5) must have added it for the sake of completeness, but since the binaries would not be available when used as a GIMP plug-in anyway, there should be no reason to include extras for a plug-in. Could you drop it and mention the reasons in the commit message?

Reply:
Done.